config/v1/techpreview.infrastructure.testsuite.yaml

```diff
@@ -528,37 +528,149 @@ tests:
           gcp:
-            clusterHostedDNS: "Enabled"
+            cloudLoadBalancerConfig:
+              dnsType: PlatformDefault
+          type: GCP
+    expectedStatusError: "status.platformStatus.gcp.cloudLoadBalancerConfig.dnsType: Invalid value: \"string\": dnsType is immutable once set"
+  - name: Should not accept non-IP address values for Load Balancer IPs
+    initial: |
+      apiVersion: config.openshift.io/v1
+      kind: Infrastructure
+      spec:
+        platformSpec:
+          gcp: {}
+          type: GCP
+      status:
+        controlPlaneTopology: HighlyAvailable
+        infrastructureTopology: HighlyAvailable
+        platform: GCP
+        platformStatus:
+          gcp: 
+            cloudLoadBalancerConfig:
+              dnsType: ClusterHosted
+              clusterHosted:
+                apiIntLoadBalancerIPs:
+                - 10.10.10.20
+          type: GCP
     updated: |
       apiVersion: config.openshift.io/v1
       kind: Infrastructure
-      spec: {}
+      spec:
+        platformSpec:
+          type: GCP
+          gcp: {}
       status:
+        controlPlaneTopology: HighlyAvailable
+        infrastructureTopology: HighlyAvailable
         platform: GCP
         platformStatus:
+          gcp:
+            cloudLoadBalancerConfig:
+              dnsType: ClusterHosted
+              clusterHosted:
+                apiIntLoadBalancerIPs:
+                - 10.10.10.20
+                - not-an-ip-address
           type: GCP
+    expectedStatusError: "status.platformStatus.gcp.cloudLoadBalancerConfig.clusterHosted.apiIntLoadBalancerIPs[1]: Invalid value: \"not-an-ip-address\": status.platformStatus.gcp.cloudLoadBalancerConfig.clusterHosted.apiIntLoadBalancerIPs[1] in body should match '(^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$)|(^s*((([0-9A-Fa-f]{1,4}:){7}([0-9A-Fa-f]{1,4}|:))|(([0-9A-Fa-f]{1,4}:){6}(:[0-9A-Fa-f]{1,4}|((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3})|:))|(([0-9A-Fa-f]{1,4}:){5}(((:[0-9A-Fa-f]{1,4}){1,2})|:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3})|:))|(([0-9A-Fa-f]{1,4}:){4}(((:[0-9A-Fa-f]{1,4}){1,3})|((:[0-9A-Fa-f]{1,4})?:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){3}(((:[0-9A-Fa-f]{1,4}){1,4})|((:[0-9A-Fa-f]{1,4}){0,2}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){2}(((:[0-9A-Fa-f]{1,4}){1,5})|((:[0-9A-Fa-f]{1,4}){0,3}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){1}(((:[0-9A-Fa-f]{1,4}){1,6})|((:[0-9A-Fa-f]{1,4}){0,4}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(:(((:[0-9A-Fa-f]{1,4}){1,7})|((:[0-9A-Fa-f]{1,4}){0,5}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:)))(%.+)?s*)'"
+  - name: Should not accept update when API-Int Load Balancer IP addresses are missing
```

Comment:
This validation is the wrong error coming out here, you need to make the APIInt field required (as per suggestions below) and not validate it within the CEL validation on the union itself